Providers/Scripts/2.4x-2.5x/Scripts/nxOMSContainers.py

```diff
@@ -0,0 +1,295 @@
+#!/usr/bin/env python
+#============================================================================
+# Copyright (C) Microsoft Corporation, All rights reserved.
+#============================================================================
+
+from contextlib import contextmanager
+
+import os
+import sys
+import imp
+import re
+import pwd
+import grp
+import codecs
+import shutil
+
+protocol = imp.load_source('protocol', '../protocol.py')
+nxDSCLog = imp.load_source('nxDSCLog', '../nxDSCLog.py')
+
+LG = nxDSCLog.DSCLog
+
+MODULE_RESOURCE_DIR ='/opt/microsoft/omsconfig/modules/nxOMSContainers/DSCResources/MSFT_nxOMSContainersResource/containers'
+
+PLUGIN_PATH = '/opt/microsoft/omsagent/plugin/'
+CONF_PATH = '/etc/opt/microsoft/omsagent/<WorkspaceID>/conf/omsagent.d'
+REG_PATH = '/etc/opt/omi/conf/omiregister/root-cimv2'
+LIB_PATH = '/opt/microsoft/docker-cimprov/lib'
+YAML_PATH = '/var/opt/microsoft/docker-cimprov/state'
+STATE_DIR_PATH = '/var/opt/microsoft/omsagent/<WorkspaceID>/state'
+
+class IOMSAgent:
+    def restart_oms_agent(self):
+        pass
+    
+class OMSAgentUtil(IOMSAgent):
+    def restart_oms_agent(self, WorkspaceID):
+        wsId = WorkspaceID
+        if wsId is None:
+            wsId = ''
+        if os.system('sudo /opt/microsoft/omsagent/bin/service_control restart %s' %(wsId)) == 0:
+            return True
+        else:
+            LOG_ACTION.log(LogType.Error, 'Error restarting omsagent for workspace ' + wsId)
+            return False
+
+class TestOMSAgent(IOMSAgent):
+    def restart_oms_agent(self):
+        return True
+
+OMS_ACTION = OMSAgentUtil()
+
+def Set_Marshall(WorkspaceID,Ensure):
+    global CONF_PATH
+    WorkspaceID = WorkspaceID.encode('ascii', 'ignore')
+    CONF_PATH = CONF_PATH.replace('<WorkspaceID>',WorkspaceID)
+    Ensure = Ensure.encode('ascii', 'ignore')
+    return Set(WorkspaceID, Ensure)
+
+def Test_Marshall(WorkspaceID,Ensure):
+    global CONF_PATH
+    WorkspaceID = WorkspaceID.encode('ascii', 'ignore')
```

Comment:
This is copied code; I would recommend handling these lines with an `init_vars` method like nxOMSPlugin: https://github.com/Microsoft/PowerShell-DSC-for-Linux/blob/master/Providers/Scripts/2.4x-2.5x/Scripts/nxOMSPlugin.py#L59